iothub/device/src/Transport/AmqpIot/AmqpIotMessageConverter.cs

```diff
@@ -37,12 +36,17 @@ internal class AmqpIotMessageConverter
 
         public static Message AmqpMessageToMessage(AmqpMessage amqpMessage)
         {
-            if (amqpMessage == null)
-            {
-                throw Fx.Exception.ArgumentNull(nameof(AmqpMessage));
-            }
-            Stream stream = amqpMessage.BodyStream;
-            var message = new Message(stream, StreamDisposalResponsibility.Sdk);
+            Argument.AssertNotNull(amqpMessage, nameof(amqpMessage));
+
+            using var ms = new MemoryStream();
+            amqpMessage.BodyStream.CopyTo(ms);
+            amqpMessage.Dispose();
```

Comment:
we're disposing this message here but looks like we're referencing it again in L50